Automattic/wp-calypso
# Themes: Add Dialog when Switching From Retired Theme

#### Changes proposed in this Pull Request

This converts the dialog that displays when activating a theme that has a Homepage Layout to one that serves that purpose, as well as warning when switching from a retired theme. I don't think that I'm able to change a few other cases of the previous name as it appears they involve conditions outside this repo, but I don't think that's a big deal.

#### Testing instructions

I took immense care here to ensure that nothing is changed when switching to a Homepage Layout, but feel free to verify this just in case! The same dialog should still appear.

This just adds a message when switching to a retired theme. Select a retired theme through WP-Admin (eg. _Harmonic_) and attempt to activate another theme. This should appear, and clicking the primary button should activate the theme as usual. Verify that this dialog doesn't appear in other cases.

<img width="953" alt="Screenshot 2020-04-05 at 18 49 01" src="https://user-images.githubusercontent.com/43215253/78506027-14aa4600-776f-11ea-8bd7-070d24f803bc.png">

cc @lancewillett 

Fixes #40310
